<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>408</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>15</number>
          </property>
          <item>
           <widget class="QGroupBox" name="oil_weight_groupbox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Oil weight</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QDoubleSpinBox" name="weight_grams">
               <property name="maximum">
                <double>50000.000000000000000</double>
               </property>
               <property name="value">
                <double>500.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="weight_unit_label">
               <property name="text">
                <string>grams</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="water_groupbox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Water</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="water_amount_label">
               <property name="text">
                <string>Water : Lye ratio</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="water_percentage">
               <property name="suffix">
                <string/>
               </property>
               <property name="maximum">
                <double>99.900000000000006</double>
               </property>
               <property name="value">
                <double>2.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="super_fat_groupbox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Super Fat</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QDoubleSpinBox" name="super_fat">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="suffix">
                <string>%</string>
               </property>
               <property name="maximum">
                <double>50.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="fragrance_groupbox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Fragrance</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QDoubleSpinBox" name="fragrance_percent">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="suffix">
                <string>â€°</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="3,0">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="sort_by_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Sort by</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="sort_order_incdec_button">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>increasing</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="sort_by_property">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="currentText">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QListWidget" name="oil_list">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>350</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTabWidget" name="tabWidget">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab_properties">
             <property name="accessibleName">
              <string/>
             </property>
             <attribute name="title">
              <string>Properties</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="1">
               <widget class="QLabel" name="one_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Selected</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Name</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="all_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Combination</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_fatty_acids">
             <attribute name="title">
              <string>Fatty acids</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="1">
               <widget class="QLabel" name="selected_acid_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Selected</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="combination_acid_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Combination</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Name</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="1">
           <widget class="QLineEdit" name="water_amount">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Water amount</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Lye amount</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lye_amount">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableView" name="ingredient_table"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1002</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Save"/>
    <addaction name="action_Load"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_Options">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <widget class="QMenu" name="menu_Lye_type">
     <property name="title">
      <string>&amp;Lye Type</string>
     </property>
     <addaction name="action_NaOH"/>
     <addaction name="action_KOH"/>
     <addaction name="actionKOH_90"/>
    </widget>
    <widget class="QMenu" name="menu_Unit">
     <property name="title">
      <string>&amp;Unit</string>
     </property>
     <addaction name="action_Grams"/>
     <addaction name="action_Pounds"/>
     <addaction name="action_Ounces"/>
    </widget>
    <widget class="QMenu" name="menu_Water">
     <property name="title">
      <string>&amp;Water</string>
     </property>
     <addaction name="actionAs_percentage_of_oils"/>
     <addaction name="action_Lye_concentration"/>
     <addaction name="actionWater_lye_ratio"/>
    </widget>
    <addaction name="menu_Lye_type"/>
    <addaction name="menu_Unit"/>
    <addaction name="menu_Water"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Options"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Save"/>
   <addaction name="action_Load"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="action_Load">
   <property name="text">
    <string>&amp;Load</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="action_Super_Fat">
   <property name="text">
    <string>&amp;Super Fat</string>
   </property>
  </action>
  <actiongroup name="actiongroup_water_amount">
   <action name="actionAs_percentage_of_oils">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>As &amp;percentage of oils</string>
    </property>
   </action>
   <action name="action_Lye_concentration">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>&amp;Lye concentration</string>
    </property>
   </action>
   <action name="actionWater_lye_ratio">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>Water : lye &amp;ratio</string>
    </property>
   </action>
  </actiongroup>
  <actiongroup name="actiongroup_lye_type">
   <action name="action_NaOH">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>&amp;NaOH</string>
    </property>
   </action>
   <action name="action_KOH">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>&amp;KOH</string>
    </property>
   </action>
   <action name="actionKOH_90">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>KOH, &amp;90%</string>
    </property>
   </action>
  </actiongroup>
  <actiongroup name="actiongroup_unit">
   <action name="action_Grams">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>&amp;Grams</string>
    </property>
   </action>
   <action name="action_Pounds">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>&amp;Pounds</string>
    </property>
   </action>
   <action name="action_Ounces">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>&amp;Ounces</string>
    </property>
   </action>
  </actiongroup>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>sort_order_incdec_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggle_sort_order()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sort_by_property</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>sort_oils_by(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>-28</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>oil_list</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>show_current_oil_properties(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>164</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>202</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>set_weight_grams(double)</signal>
  <signal>set_weight_pounds(double)</signal>
  <signal>set_weight_ounces(double)</signal>
  <slot>toggle_sort_order()</slot>
  <slot>sort_oils_by(QString)</slot>
  <slot>show_current_oil_properties(QString)</slot>
 </slots>
</ui>
